qemu/tests/qemu_guest_agent.py

```diff
@@ -1375,6 +1373,134 @@ def setup(self, test, params, env):
             args = [params.get("gagent_serial_type"), params.get("gagent_name")]
             self.gagent_create(params, self.vm, *args)
 
+    def _fswrite_check(self, flag):
+        """
+        Get the result from vm and check if the time stamp is not continuous
+        if not continuous,guest file system is frozen.
+        """
+        # init fs status to 'thaw'
+        fs_status = "thaw"
+        file_name = "/tmp/fsfreeze_%s.txt" % flag
+        process.system("rm -rf %s" % file_name)
+        self.vm.copy_files_from("C:\\fsfreeze.txt", file_name)
+        with open(file_name, 'r') as f:
+            list_time = f.readlines()
+
+        for i in list(range(0, len(list_time))):
+            list_time[i] = list_time[i].strip('')
+
+        for i in list(range(1, len(list_time))):
+            num_d = float(list_time[i]) - float(list_time[i - 1])
+            if num_d > 8:
+                logging.info("Time stamp is not continuous,"
+                             " so the FS is frozen.")
+                fs_status = "frozen"
+                break
+        if not fs_status == flag:
+            self.test.fail("FS is not %s, it's %s." % (flag, fs_status))
+
+    @error_context.context_aware
+    def _action_before(self):
```

Comment:
This class is not only for fsfreeze/thaw, so the function name better have definite directing, like `action_after_freeze_thaw`. Actually this function has no other functionality besides start the backgroud test, so it even can be named to `background_write()`, and also it only serves gagent_check_fsfreeze so it can be included inside `gagent_check_fsfreeze()`. Same for `_action_after`

Reply:
When I design this case,just want to make it align with linux part. At present, agree with you, it's better to move _action_after and _action_before in  gagent_check_fsfreeze().